lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -370,54 +425,97 @@ def match_annotations(annotations)
             if cached
-              @cache[cache_key] = parse_pod_metadata(notice.object)
+              @pod_cache[cache_key] = parse_pod_metadata(notice.object)
+              @pod_cache_watch_updates += 1
+            else
+              @pod_cache_watch_misses += 1
             end
           when 'DELETED'
             cache_key = "#{notice.object['metadata']['namespace']}_#{notice.object['metadata']['name']}"
-            @cache.delete(cache_key)
+            @pod_cache.delete(cache_key)
+            @pod_cache_watch_deletes += 1
           else
             # Don't pay attention to creations, since the created pod may not
             # end up on this node.
+            @pod_cache_watch_ignored += 1
         end
       end
     end
 
     def start_namespace_watch
-      resource_version = @client.get_namespaces.resourceVersion
-      watcher          = @client.watch_namespaces(resource_version)
+      begin
+        resource_version = @client.get_namespaces.resourceVersion
+        watcher          = @client.watch_namespaces(resource_version)
+      rescue Exception => e
+        message = "start_namespace_watch: Exception encountered setting up namespace watch from Kubernetes API #{@apiVersion} endpoint #{@kubernetes_url}: #{e.message}"
+        if e.respond_to?(:response)
+          message += " (#{e.response})"
+        end
+        log.debug message
+        raise Fluent::ConfigError, message
+      end
+
       watcher.each do |notice|
-        puts notice
         case notice.type
           when 'MODIFIED'
             cache_key = notice.object['metadata']['name']
             cached    = @namespace_cache[cache_key]
             if cached
               @namespace_cache[cache_key] = parse_namespace_metadata(notice.object)
+              @namespace_cache_watch_updates += 1
+            else
+              @namespace_cache_watch_misses += 1
             end
           when 'DELETED'
             @namespace_cache.delete(notice.object['metadata']['name'])
+            @namespace_cache_watch_deletes += 1
           else
             # Don't pay attention to creations, since the created namespace may not
             # be used by any pod on this node.
+            @namespace_cache_watch_ignored += 1
+        end
+      end
+    end
+
+    def dump_stats
+      @curr_time = Time.now
+      return if @curr_time - @prev_time < @stats_interval
+      @prev_time = @curr_time
+      if @use_journal
+        log.info "container name match failed: #{@container_name_match_failed}, container name id missing: #{@container_name_id_missing}"
+      end
+      if @merge_json_log
+        log.info "merge json parse errors: #{@merge_json_parse_errors}"
+      end
+      if @kubernetes_url.present?
+        log.info "pod cache stats api: updates: #{@pod_cache_api_updates}, nil_not_found: #{@pod_cache_api_nil_not_found}, nil_bad_resp_payload: #{@pod_cache_api_nil_bad_resp_payload}, nil_error: #{@pod_cache_api_nil_error}"
```

Comment:
Adding a Counter class would allow you to create a `to_s` and just dump the counter

Reply:
https://github.com/balgarath/ruby-counter-class